src/infra/src/table/dashboards.rs

```diff
@@ -168,22 +161,25 @@ pub async fn put(
         .map(|d| d.to_owned());
     let version = dashboard.version;
     let created_at_depricated = dashboard.created_at_deprecated();
-
     let data = inner_data_as_json(dashboard)?;
 
-    let (folder_m, mut dash_am) = match get_model(client, org_id, folder_id, &dashboard_id).await? {
-        None => {
-            // Destination folder does not exist so the dashboard can neither be
-            // created nor updated.
-            Err(errors::PutDashboardError::FolderDoesNotExist)
-        }
-        Some((folder_m, Some(dash_m))) => {
+    let client = ORM_CLIENT.get_or_init(connect_to_orm).await;
+    let tx = client.begin().await?;
+
+    let Some(folder_m) = super::folders::get_model(&tx, org_id, folder_id).await? else {
+        // Destination folder does not exist so the dashboard can neither be
+        // created nor updated.
+        return Err(errors::PutDashboardError::FolderDoesNotExist.into());
+    };
```

Comment:
Since the `get_folder_model` method no longer takes `folder_id` as a parameter, we will get the folder model first to make sure the destination folder actually exists. Since this now happens in a separate query from the query to try to get the dashboard, we perform the "get folder" and "get dashboard" operations in a transaction.

Reply:
We probably should have doing the "get dashboard" and "save dashboard" operations in a transaction even before this PR.